src/js_native_api_v8.h

```diff
@@ -82,8 +82,15 @@ struct napi_env__ {
     return v8::Just(true);
   }
 
-  template <typename T, typename U>
-  void CallIntoModule(T&& call, U&& handle_exception) {
+  static inline void
+  HandleThrow(napi_env env, v8::Local<v8::Value> value) {
+    env->isolate->ThrowException(value);
+  }
+  using ThrowHandler = std::function<void(napi_env, v8::Local<v8::Value>)>;
+
+  template <typename T>
+  inline void CallIntoModule(T&& call,
+                             ThrowHandler handle_exception = HandleThrow) {
```

Comment:
Is there any reason that `handle_exception` is no longer a second template argument? `std::function` incurs both memory and CPU overheads, and even though they are probably small, I would try to avoid that.

(If you must use `std::function` for some reason, it would be nice not to wrap `HandleThrow` as the default argument each time.)

Reply:
I couldn't get it to build that way 😀 I used `std::function` because its signature supports `[&]`. I'll try and see if I can do it differently.

Reply:
@gabrielschulhof This seems to work for me:

```diff
diff --git a/src/js_native_api_v8.h b/src/js_native_api_v8.h
index 29c341c6d3f1..b2f0889f0f61 100644
--- a/src/js_native_api_v8.h
+++ b/src/js_native_api_v8.h
@@ -86,11 +86,10 @@ struct napi_env__ {
   HandleThrow(napi_env env, v8::Local<v8::Value> value) {
     env->isolate->ThrowException(value);
   }
-  using ThrowHandler = std::function<void(napi_env, v8::Local<v8::Value>)>;
 
-  template <typename T>
+  template <typename T, typename U = decltype(HandleThrow)>
   inline void CallIntoModule(T&& call,
-                             ThrowHandler handle_exception = HandleThrow) {
+                             U&& handle_exception = HandleThrow) {
     int open_handle_scopes_before = open_handle_scopes;
     int open_callback_scopes_before = open_callback_scopes;
     napi_clear_last_error(this);
```

Reply:
Cool 👍 This is actually more succinct than what I had 🙂